File: backend/src/main/java/com/backend/global/config/AsyncConfig.java
package com.backend.global.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
import java.util.concurrent.Executor;

@Configuration
@EnableAsync
public class AsyncConfig {

    @Bean(name = "feedbackExecutor")
    public Executor feedbackExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        executor.setCorePoolSize(5);
        executor.setMaxPoolSize(10);
        executor.setQueueCapacity(100);
        executor.setThreadNamePrefix("feedback-");
        executor.initialize();
        return executor;
    }

    @Bean(name = "mailExecutor")
    public Executor mailExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        executor.setCorePoolSize(3);       // 기본 스레드 수
        executor.setMaxPoolSize(10);       // 최대 스레드 수
        executor.setQueueCapacity(50);     // 큐 용량
        executor.setThreadNamePrefix("MailExecutor-"); // 스레드 이름 prefix
        executor.initialize();
        return executor;
    }
}

File: backend/src/main/java/com/backend/global/config/AsyncConfig.kt
package com.backend.global.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor

@Configuration
@EnableAsync
class AsyncConfig {

    @Bean(name = ["feedbackExecutor"])
    fun feedbackExecutor(): Executor = ThreadPoolTaskExecutor().apply {
        corePoolSize = 5
        maxPoolSize = 10
        queueCapacity = 100
        setThreadNamePrefix("feedback-")
        initialize()
    }

    @Bean(name = ["mailExecutor"])
    fun mailExecutor(): Executor = ThreadPoolTaskExecutor().apply {
        corePoolSize = 3       // 기본 스레드 수
        maxPoolSize = 10       // 최대 스레드 수
        queueCapacity = 50     // 큐 용량
        setThreadNamePrefix("MailExecutor-")
        initialize()
    }
}
